sympy/stats/rv.py

```diff
@@ -275,6 +275,19 @@ def _hashable_content(self):
     def free_symbols(self):
         return {self}
 
+class RandomIndexedSymbol(RandomSymbol):
+
+    def __new__(cls, idx_obj, pspace=None):
+        if not isinstance(idx_obj, Indexed):
+            raise TypeError("An indexed object is expected not %s"%(idx_obj))
+        return Basic.__new__(cls, idx_obj, pspace)
+
+    symbol = property(lambda self: self.args[0])
+    name = symbol
+    key = property(lambda self: self.symbol.args[1])
+
+    def _hashable_content(self):
+        return self.pspace, self.name, self.key
```

Comment:
Why do you need to define this and why isn't it returning the same as args?

Reply:
`RandomSymbol` defines `_hashable_content`, [here](https://github.com/sympy/sympy/blob/master/sympy/stats/rv.py#L271). So to override that I defined it in `RandomIndexedSymbol`. 
Sure, I will make it return the same as `args`. 
PS - By the way, is this a constraint in SymPy that `_hashable_content` should return `args`.

Reply:
It shouldn't be defined in `RandomSymbol` either I think.

There is no point in defining hashable_content if it just returns the args. It is already defined that way in Basic. The only reason to define hashable_content is if you want it to do something different.

Reply:
> It shouldn't be defined in `RandomSymbol` either I think.

Okay, I will remove it from `RandomSymbol` and `RandomIndexedSymbol`.

> There is no point in defining hashable_content if it just returns the args. It is already defined that way in Basic. The only reason to define hashable_content is if you want it to do something different.

Thanks for the explanation.